src/regexp.toit

```diff
@@ -1271,30 +1271,34 @@ class Capture_ extends MiniExpAst_:
 interface Match:
   /// Returns the regular expression that created this match.
   pattern -> RegExp
-  /// Returns the input string that was matched by the regexp.
+
+  /// Returns the input string in which a match was found.
   input -> string
+
+  /// Returns the substring of the input that matched the regexp.
+  matched -> string
+
   /// Returns the start index of the match in the string.
-  start -> int
+  index -> int
+
+  /// Returns the end index of the match in the string.
+  end_index -> int
+
   /**
   Returns the start index of the nth capture in the string.
-  The 0th capture is the entire match of the regexp.
+  The 0th capture is the index of the match of the whole regexp.
   */
-  start index/int -> int?
+  index index/int -> int?
+
   /**
   Returns the end index of the nth capture in the string.
-  The 0th capture is the entire match of the regexp.
-  */
-  end -> int
-  ///Returns the end index of the match in the string
-  end index/int -> int?
-  /**
-  Returns the string captured by the nth capture.
-  The 0th capture is the entire match of the regexp.
+  The 0th capture is the index of the match of the whole regexp.
   */
-  group index/int -> string?
+  end_index index/int -> int?
+
   /**
   Returns the string captured by the nth capture.
-  The 0th capture is the entire match of the regexp.
+  The 0th capture is the entire substring that matched the regexp.
```

Comment:
Is there a good reason to keep that behavior?
What's the advantage of having the full match in the list?

Reply:
This just aligns very well with what people expect from other regexp engines.